src/components/Layout/Header/NavBar/Native/BackupPassphraseModal/BackupPassphraseTest.tsx

```diff
@@ -0,0 +1,165 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
+import {
+  Button,
+  IconButton,
+  ModalBody,
+  ModalCloseButton,
+  ModalHeader,
+  Tag,
+  Wrap,
+} from '@chakra-ui/react'
+import * as native from '@shapeshiftoss/hdwallet-native'
+import { Vault } from '@shapeshiftoss/hdwallet-native-vault'
+import * as bip39 from 'bip39'
+import range from 'lodash/range'
+import shuffle from 'lodash/shuffle'
+import slice from 'lodash/slice'
+import uniq from 'lodash/uniq'
+import { useCallback, useEffect, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router'
+import { SlideTransition } from 'components/SlideTransition'
+import { RawText, Text } from 'components/Text'
+
+import { BackupPassphraseRoutes } from './BackupPassphraseCommon'
+
+const Revocable = native.crypto.Isolation.Engines.Default.Revocable
+const revocable = native.crypto.Isolation.Engines.Default.revocable
+
+const TEST_COUNT_REQUIRED = 3
+
+const ordinalSuffix = (n: number) => {
+  return ['st', 'nd', 'rd'][((((n + 90) % 100) - 10) % 10) - 1] || 'th'
+}
+
+type TestState = {
+  targetWordIndex: number
+  randomWords: string[]
+  correctAnswerIndex: number
+}
+
+export const BackupPassphraseTest = ({ vault }: { vault: Vault | null }) => {
+  const translate = useTranslate()
+  const history = useHistory()
+  const [testState, setTestState] = useState<TestState | null>(null)
+  const [invalidTries, setInvalidTries] = useState<number[]>([])
+  const [testCount, setTestCount] = useState<number>(0)
+  const [revoker] = useState(new (Revocable(class {}))())
+  const [shuffledNumbers] = useState(slice(shuffle(range(12)), 0, TEST_COUNT_REQUIRED))
```

Comment:
Abstract away to `initialShuffledNumbersState` and/or disable eslint for it? Clojure-style expressions are hard to read when one-lined